### PR TITLE
graphics/wayland : first attempt at using FreeBSD port as base

### PR DESCRIPTION
**Pull request by daftaupe**

### CODE REVIEW

**Comment by tuxillo** [comment]
So the status of this PR is:

- It builds fine
- Requires to bring additional patches from the old newport

Is this correct?

**Comment by daftaupe** [comment]
> So the status of this PR is:
> 
>     * It builds fine
> 
>     * Requires to bring additional patches from the old newport
> 
> 
> Is this correct?

@tuxillo, yes that's exactly the status of the PR

File: ports/graphics/wayland/dragonfly/patch-src_wayland-os.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
++#if defined(__FreeBSD__) || defined(__DragonFly__)
+ int
+ wl_os_socket_peercred(int sockfd, uid_t *uid, gid_t *gid, pid_t *pid)
+ {
+@@ -87,7 +89,7 @@
+ 	struct xucred ucred;
+ 
+ 	len = sizeof(ucred);
+-	if (getsockopt(sockfd, SOL_LOCAL, LOCAL_PEERCRED, &ucred, &len) < 0 ||
++	if (getsockopt(sockfd, 0, LOCAL_PEERCRED, &ucred, &len) < 0 ||
```
Comment (daftaupe) [inline ports/graphics/wayland/dragonfly/patch-src_wayland-os.c:38]: really not sure about that one, I took the 0 from devel/basu/dragonfly/patch-src_basic_socket-util.c

File: ports/graphics/wayland/dragonfly/patch-src_wayland-server.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ #include <fcntl.h>
+-#include <sys/eventfd.h>
+ #include <sys/file.h>
+ #include <sys/stat.h>
+ 
+@@ -1075,10 +1074,6 @@
+ 		return NULL;
+ 	}
+ 
+-	display->terminate_efd = eventfd(0, EFD_CLOEXEC | EFD_NONBLOCK);
```
Comment (daftaupe) [inline ports/graphics/wayland/dragonfly/patch-src_wayland-server.c:15]: removing that whole block is probably not the best idea ever, but no clue how to replace it
removing it made me remove the err_eventfd too

**Pull request merged**